Fix3687 - card border white edges

corvinsz opened the pull request:
I added a `Card` showcasing the linked bug in the demo app (make sure to remove it if this PR gets merged).
When removing the `Background="{TemplateBinding Background}"` of the nested `Card` in the `AdornerDecorator`, the bug seems to be fixed.

Before:
![image](https://github.com/user-attachments/assets/96066522-0645-4586-a7c0-7b02f3e0d56c)


After:
![image](https://github.com/user-attachments/assets/327d1043-c781-4999-82f7-df9408c3acab)

As shown with the text, ClearType is still maintained.

I'm not sure why there are 2 Borders in the AdornerDecorator in the first place, so any input on if this is breaking something is welcome.

Edit: this possibly fixes #3687 
Display: 1080p monitor @150% scaling

Keboo reviewed (changes requested):
Great working tracking it down, although I am not sure this is the correct fix. That background color is there to effectively provide an anchor for the drop shadow. Without that background binding the elevation no longer shows properly 
(It becomes most obvious in light mode with something like this: `<materialDesign:Card materialDesign:ElevationAssist.Elevation="Dp24">`).

Effectively the layout rounding and the clipping on the corners is what appears to have the background showing through. This does not apply the outlined card style as that one does not have the elevation (drop shadow added to it).

I think a better fix will be to simply apply a small Margin (something like `Margin="1") or simply adding a BorderThickness (as you pointed out in the issue) to inset this adorner. Both of these effectively shrink the size of that border just enough that any layout rounding wont be visible.

I also wanted to thank you for all of the work and help you have been doing on this project recently. It is appreciated.

corvinsz commented:
* Added back the `Background="{TemplateBinding Background}"`
* Added `Margin="1"` to inset the Border (as you suggested)
Result:
![image](https://github.com/user-attachments/assets/201e9a23-29c8-4c50-9de3-afc3bb3849e8)

As it can be seen in the image above, the white edges/corners are still visible with the applied changes.
This can be circumvented by applying the `CornerRadius` to the inner `Border`, which has already been done to the outer `Border`.

To be honest this still feels kinda hacky, but it appears to fix the issue:
![image](https://github.com/user-attachments/assets/7f7e7951-0e03-4672-b979-1cd22a344d17)


P.S.: I want you thank you for your work. I really enjoy your live streams (or youtube videos rather, because I'm almost never up intime because I'm from the EU), they have taught me alot!
